lib/rules/no-prototype-builtins.js

```diff
@@ -59,7 +95,38 @@ module.exports = {
                     messageId: "prototypeBuildIn",
                     loc: callee.property.loc,
                     data: { prop: propName },
-                    node
+                    node,
+                    suggest: [
+                        {
+                            messageId: "callObjectPrototype",
+                            data: { prop: propName },
+                            fix(fixer) {
+
+                                /*
+                                 * a call after an optional chain (e.g. a?.b.hasOwnProperty(c))
+                                 * must be rewritten manually because the call can be short-circuited
+                                 */
+                                const isFixable = !isAfterOptional(node);
```

Comment:
```suggestion
                                if(isAfterOptional(node)) {
                                    return null;
                                }
                                
```

We can return `null` right away here, there's no need to calculate the fix if it isn't going to be returned.